From edf83a49f82a4a98b30c2083e59e8dcfc802f8dd Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 19 Mar 2015 17:45:21 +0000
Subject: [PATCH 0574/1000] MANTIS 0035986 extension: appControl: some
 applications are visible when CLA is active

massai_systools.lib
V1.0.0.27

- added methods mSysWindow::setTopMost, mSysWindow::resetTopMost




git-svn-id: svn://localhost/SelfServiceCommon/trunk@834 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h   | 1 +
 SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp  | 2 +-
 .../Massai/cpp/Systools/src/mSysWindowW32.cpp            | 9 +++++++++
 3 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h b/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h
index b46ae6c2..f7654db0 100644
--- a/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h
+++ b/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h
@@ -27,6 +27,7 @@ namespace mSysWindow
       virtual void minimize()= 0;
       virtual void showDefault()= 0;
       virtual void setTopMost()= 0;
+      virtual void resetTopMost()= 0;
       virtual void show()= 0;
       virtual void hide()= 0;
       virtual void close()= 0;
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
index 4f16f8cd..5d8bd034 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
@@ -17,7 +17,7 @@ massai_systools.lib
 \version 1.0.0.27
 \li 19.03.2015
 \li extension SteT MANTIS 0035986: appControl: some applications are visible when CLA is active
-- added methods mSysWindow::setTopMost
+- added methods mSysWindow::setTopMost, mSysWindow::resetTopMost
 
 \version 1.0.0.26
 \li 27.02.2015
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp
index 2824ac32..2bdee5a1 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp
@@ -37,6 +37,7 @@ namespace
       virtual std::string getProcessId();
       virtual bool isClosed();
       virtual void setTopMost();
+      virtual void resetTopMost();
 
       DWORD getPID() const;
     private:
@@ -136,6 +137,14 @@ namespace
     }
   }
 
+  void WindowW32::resetTopMost()
+  {
+    if(!SetWindowPos(m_hwnd, HWND_NOTOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE))
+    {
+      traceLastError(MinorError_HERE, "SetWindowPos", mSysError::getErrorCode());
+    }
+  }
+
   bool WindowW32::switchToWindow()
   {
     bool result = true;
-- 
2.41.0.windows.1

